modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2387,6 +2387,17 @@ def apply(self, func, axis, *args, **kwargs):
         else:
             return self._callable_func(func, axis, *args, **kwargs)
 
+    def apply_on_series(self, func, *args, **kwargs):
+        args = try_cast_to_pandas(args)
+        kwargs = try_cast_to_pandas(kwargs)
+
+        assert self.is_series_like()
+        return self.__constructor__(
+            self._modin_frame.apply_full_axis(
+                1, lambda df: df.squeeze(axis=1).apply(func, *args, **kwargs)
```

Comment:
Why do we have to use `apply_full_axis` instead of `map` as it was before? That significantly slows down the things.

Reply:
I tried using map but things were breaking. I can try it again and look into it.

Reply:
Yes, please look into it.

Reply:
@YarShev we can't use `map` because I think the function doesn't assume that the dimensions of the DataFrame will not change. That's why I went with `apply_full_axis` here since we may need to recompute some metadata/indexing.

Reply:
So probably for now we only can go with `apply_full_axis`. That fixes the issue but the perf is not pretty good. As to the perf, we can thing of it as part of a separate issue.